conwayconsole/Program.cs

```diff
@@ -9,140 +9,189 @@
+                        {
+                            Console.WriteLine("Could not read from file");
+                            return;
+                        }
+                    }
+                    else
+                    {
+                        // create board based on settings
+                        var random = options.Seed.HasValue ? new Random(options.Seed.Value) : new Random();
+                        double density = 1 - Math.Clamp(options.Density, 0, 1);
+                        initialBoard = new Board(options.BoardWidth, options.BoardHeight);
+                        initialBoard.Randomise(random, density);
+                    }
+
+                    KeyMonitor.Movement += (_, args) =>
+                    {
+
+                        window.X = Math.Min(Math.Max(0, args.Horizontal + window.X), initialBoard.Width - window.Width);
+                        window.Y = Math.Min(Math.Max(0, args.Vertical + window.Y), initialBoard.Height - window.Height);
+
+                    };
+
+                    bool save = false;
+                    KeyMonitor.Save += (_, args) => save = true;
+
+                    var game = new Game(initialBoard, StandardEvolution.Instance);
+                    var builder = new StringBuilder();
+
+                    if (!options.HideDisplay)
+                    {
+                        Console.Clear();
+                        Console.CursorVisible = false;
+                    }
+
+                    bool stop = false;
+                    DateTime lastLoopTime = DateTime.UtcNow;
+                    for (IReadableBoard board = initialBoard;
+                      !(stop || cts.IsCancellationRequested) && (options.MaxGenerations >= game.Generation);
+                      board = game.Turn(out stop))
+                    {
+                        if (!options.HideDisplay)
+                        {
+                            await Console.Out.WriteLineAsync(board.ToConsoleString(window, builder, getValueForColour));
+                            DateTime now = DateTime.UtcNow;
+                            TimeSpan elapsed = now.Subtract(lastLoopTime);
+                            TimeSpan delay = TimeSpan.FromMilliseconds(options.Delay).Subtract(elapsed);
+
+                            await Console.Out.WriteLineAsync($"{BoardConsoleExtensions.cHome}({board.Width}x{board.Height}) ({window.Width}x{window.Height}) ({window.X}, {window.Y})");
+                            if (delay > TimeSpan.Zero)
+                            {
+                                if (delay > TimeSpan.Zero)
+                                {
+                                    await Task.Delay(delay, cts.Token);
+                                }
+
+                                lastLoopTime = DateTime.UtcNow;
+                            }
+                        }
+
+                        if (save)
+                        {
+                            await Save(board);
```

Comment:
This is doing the save during the loop, which means the cursor and colour has not been reset. At one point, I had moved the code to after the loop and the `finally` so that it would not have that problem, but this seems to have been undone.